language-design.md

```diff
@@ -59,7 +83,13 @@ What does the post on grayscale tell us about the process of API design?
 
 **Response**
 
+Lea Verou’s forum post on grayscale is the embodiment of several important design paradigms explored in Bloch’s API design paper. To elaborate, this forum post was created by Lea Verou to ask users to vote on the best name for an API function she was designing. Specifically, the function name was supposed to describe a routine that sets a color to an appropriate shade of gray when provided a lightness and optional alpha parameter.
+ 
+By asking her users to vote on a function name, Verou is clearly in compliance with Bloch’s suggestion that “Names matter,” and “API design is not a solitary activity” [Bloch, 2006, 506]. For instance, if the function names or her users did not matter, why would the developer ever go through the trouble of poling users for something so trivial? It is because, like a DSL, a good API strives to be a simplification of functionality that is intuitive for users. Therefore, a good API should not be what is easiest for the designer to implement and remember, but should conform to the needs of the user base.
 
+Furthermore, assuming Verou will name her function in correspondence to the popular vote, by quantifying the user’s opinions, Verou is also complying with Bloch’s advice that, “You can’t please everyone so aim to displease everyone equally” [Bloch, 2006, 506]. While this advice is a little bit of a joke, the essential take away is that _any_ decision made by a developer will likely benefit one user while inconveniencing another – therefore, since every decision has both good and bad consequences, a developer should strive to benefit the most users while catering to no individual in specific. For instance, none of the proposed function names in this post are illogical. In fact, they’re all _good_ since they make sense in their respective contexts and follow Bloch’s advice of “…the principle of least astonishment,” where every “…method should do the least surprising thing it could, given its name” [Bloch, 2006, 506]. Verou even shows concern for overloading the rgb() and rgba() functions for this purpose similar to Bloch’s advice of  “Overload with care.” [Bloch, 2006, 507]. Therefore, by voting Verou is essentially attempting to displease the fewest number of users, or in other words, “displease everyone equally” [Bloch, 2006, 506].
+
```

Comment:
I think it's amazing how a thing so seemingly simple as a name can require such concern, sometimes due to the accidents of history such as pre-existing function names.